src/tests/Model/ResultSummaryCreatorTests.cs

```diff
@@ -0,0 +1,162 @@
+﻿using NUnit.Framework;
+using System;
+
+namespace NUnit.Gui.Model
+{
+    [TestFixture]
+    public class ResultSummaryCreatorTests
+    {
+        [Test]
+        public void WhenResultNodeIsNotForTestRunExceptionIsThrown()
+        {
+            Assert.That(() => CreateResultSummary("<anything-other-than-test-run/>"), Throws.InstanceOf<InvalidOperationException>());
+
+            Assert.That(() => CreateResultSummary("<test-run/>"), Throws.Nothing);
+        }
+        
+        [Test]
+        public void WhenResultIsNotSpecified_PassedIsDefault()
+        {
+            var summary = CreateResultSummary("<test-run/>");
+
+            Assert.That(summary.OverallResult, Is.EqualTo("Passed"));
+
+            summary = CreateResultSummary("<test-run result='Failed'/>");
+
+            Assert.That(summary.OverallResult, Is.EqualTo("Failed"));
+        }
+
+        [Test]
+        public void WhenDurationIsNotSpecified_ZeroIsDefault()
+        {
```

Comment:
Perhaps write this as two separate tests? Then we can also use a TestCase for duration to test other durations.